keyhive sofle rgb: fix configuration. (#15339)
qmk/qmk_firmware: keyboards/sofle/keyhive/readme.md

# Keyhive Sofle Keyboard

![SofleKeyboard version 2.1 RGB Keyhive](https://i.imgur.com/WH9OoWuh.jpg)

Sofle is 6×4+5 keys column-staggered split keyboard. Based on Lily58, Corne and Helix keyboards.

For details about the keyboard design, refer to Josef's blog: [Sofle Keyboard - a split keyboard based on Lily58 and Crkbd](https://josef-adamcik.cz/electronics/let-me-introduce-you-sofle-keyboard-split-keyboard-based-on-lily58.html)

Build guide: [Keyhive Sofle RGB build guide](https://github.com/keyhive/build_guides/blob/master/docs/keyboards/sofle-rgb.md)

* Keyboard Maintainer: [Winder](https://github.com/winder)
* Hardware Supported: SofleKeyboard V2.1 RGB PCB, ProMicro / Elite-C
* Hardware Availability: [Keyhive](https://keyhive.xyz/shop/sofle)

### Acknowledgements

> * Solartempest - the image on this page and most of the code is either copied directly or inspired by their for. [Solartempest's fork.](https://github.com/solartempest/qmk_firmware/tree/master/keyboards/solartempest/sofle).

Author: ```suggestion
* Solartempest - the image on this page and most of the code is either copied directly or inspired by their fork. [Solartempest's fork.](https://github.com/solartempest/qmk_firmware/tree/master/keyboards/solartempest/sofle).
```

* [Keyhive fork](https://github.com/keyhive/qmk_firmware) defined all of the board settings.

# Supported Keymaps
* [keyhive_via](../keymaps/keyhive_via/readme.md) - Includes special support for remapping encoders with VIA.
* Backwards compatibility for other keymaps is included but has not been tested.

# VIA Support
As of 1.3.1, the VIA tool does not support Keyhive/Sofle V2 out of the box.
See [keyhive_via](../keymaps/keyhive_via/readme.md) for details about configuring and using VIA.

# Compiling

Make example for this keyboard:

> ```sh
> qmk compile -kb sofle/keyhive -km keyhive_via
> ```

Reviewer: ```suggestion

    make sofle/keyhive:default
```

Author: Are `qmk compile` and `qmk flash` deprecated? It was really convenient that you could run those commands from any directory in the repo.

Reviewer: No, but the make commands are what's currently in the template. Eventually that will change.

Author: Ok, keeping things consistent until a migration sounds like it would make things easier.

Author: I just noticed that you changed the keymap here to default. Almost all of my testing was done with the "keyhive_via" keymap, which was tailor-made for this version of the board. So I wanted to use that as the default.

Reviewer: Again, per the template the "example commands" should use the `default` keymap. They are, after all, examples, not a full guide on compiling a particular board.

Author: Thanks, updated everything to use the default keymap!

## Flashing

Flash using the correct command below (or use QMK Toolbox). These commands can be mixed if, for example, you have an Elite C on the left and a pro micro on the right.

> Press reset button on he keyboard when asked.

Reviewer: ```suggestion
Press reset button on the keyboard when asked.
```

Disconnect the first half, connect the second one and repeat the process.

> ```sh
> # for pro micro-based builds
> qmk flash -kb sofle/keyhive -km keyhive_via -bl avrdude-split-left
> qmk flash -kb sofle/keyhive -km keyhive_via -bl avrdude-split-right
> 
> # for Elite C or dfu bootloader builds
> qmk flash -kb sofle/keyhive -km keyhive_via -bl dfu-split-left
> qmk flash -kb sofle/keyhive -km keyhive_via -bl dfu-split-right
> ```

Reviewer: ```suggestion
    # for Pro Micro-based builds
    make sofle/keyhive:default:avrdude-split-left
    make sofle/keyhive:default:avrdude-split-right

    # for Elite C or DFU bootloader builds
    make sofle/keyhive:default:dfu-split-left
    make sofle/keyhive:default:dfu-split-right
```
